qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py

```diff
@@ -542,30 +656,265 @@ def _compute_excitation_energies(
+                for aux_name in pre_tap_aux_observables.keys():
+                    aux_operators_eigenvalues[indice][aux_name] = aux_measurements.get(
+                        (aux_name, indice[0], indice[1]), (0.0, {})
+                    )
+
+            # 5. Format transition_amplitudes
+            indices_offdiag = np.triu_indices(size + 1, k=1)
+            indices_offdiag_as_list = list(zip(indices_offdiag[0], indices_offdiag[1]))
+            for indice in indices_offdiag_as_list:
+                transition_amplitudes[indice] = {}
+                for aux_name in pre_tap_aux_observables.keys():
+                    transition_amplitudes[indice][aux_name] = aux_measurements.get(
+                        (aux_name, indice[0], indice[1]), (0.0, {})
+                    )
+
+        return aux_operators_eigenvalues, transition_amplitudes
+
+    def _build_qeom_result(
+        self,
+        problem,
+        groundstate_result,
+        expansion_coefs,
+        energy_gaps,
+        h_mat,
+        s_mat,
+        h_mat_std,
+        s_mat_std,
+        aux_operators_eigenvalues,
+        transition_amplitudes,
+        gammas_square,
+    ) -> ElectronicStructureResult:
+
+        qeom_result = QEOMResult()
+        qeom_result.ground_state_raw_result = groundstate_result.raw_result
+        qeom_result.expansion_coefficients = expansion_coefs
+        qeom_result.excitation_energies = energy_gaps
+        qeom_result.h_matrix = h_mat
+        qeom_result.s_matrix = s_mat
+        qeom_result.h_matrix_std = h_mat_std
+        qeom_result.s_matrix_std = s_mat_std
+        qeom_result.gamma_square = gammas_square
+
+        qeom_result.aux_operators_evaluated = list(aux_operators_eigenvalues.values())
+        qeom_result.transition_amplitudes = transition_amplitudes
+
+        groundstate_energy_reference = groundstate_result.eigenvalues[0]
+        excited_eigenenergies = energy_gaps + groundstate_energy_reference
+        qeom_result.eigenvalues = np.append(
+            groundstate_result.eigenvalues[0], excited_eigenenergies
+        )
+
+        qeom_result.eigenstates = np.array([])
+
+        eigenstate_result = EigenstateResult.from_result(qeom_result)
+        result = problem.interpret(eigenstate_result)
+
+        return result
+
+
+class QEOMResult(EigensolverResult):
     """The results class for the QEOM algorithm."""
 
     def __init__(self) -> None:
         super().__init__()
         self._ground_state_raw_result = None
         self._excitation_energies: Optional[np.ndarray] = None
         self._expansion_coefficients: Optional[np.ndarray] = None
-        self._m_matrix: Optional[np.ndarray] = None
-        self._v_matrix: Optional[np.ndarray] = None
-        self._q_matrix: Optional[np.ndarray] = None
-        self._w_matrix: Optional[np.ndarray] = None
-        self._v_matrix_std: float = 0.0
-        self._q_matrix_std: float = 0.0
-        self._w_matrix_std: float = 0.0
```

Comment:
Unfortunately I don't think we can simply remove these result objects without formally deprecating them..
Can we extract these back from the `H` and `S` matrix to avoid making the actual runtime code ugly and instead extract the logic of finding these into this result object to formally deprecate these?

Reply:
I'm fine with extracting M,V,Q,W from H and S in the result class. This seems like an easy fix